runtime-sdk/modules/contracts/src/results.rs

```diff
@@ -148,6 +148,15 @@ fn process_subcalls<Cfg: Config, C: TxContext>(
                 // preconfigured the amount of available gas.
                 <C::Runtime as Runtime>::Core::use_tx_gas(ctx, result.gas_used)?;
 
+                // Forward any emitted event tags.
+                ctx.emit_etags(result.state.events);
```

Comment:
The reason is that some callers don't want these to be emitted immediately in `subcall::call` but need to handle it themselves (e.g. because they are managing state outside the usual SDK contexts). The order in this case should be the same as previously it was at the end of `subcall::call`.

Reply:
The new EVM subcall precompile, because the EVM stack executor has its own notion of state (e.g. the substate structure) and does not use our contexts.

Reply:
that one currently doesn't give any access to these events right? do the events just disappear?

Reply:
No, for EVM the events are stored in the relevant substate (similar to our context) and if substate is committed they are propagated further.

If the root substate is comitted the events are emitted by the transaction, the same as EVM logs. See the `Backend::apply` function.

Reply:
if a wasm contract subcalled an evm contract which subcalled some third thing, would that start emitting events once that third thing returned and the evm contract does `apply` (does it do `apply` at that point?)?

Reply:
Contexts are nested, so each subcall (assuming it commits) propagates events to the parent.

In your example, assuming everything commits, the third subcall propagates events into the evm substate, evm apply propagates events into the contracts subcall context and the contracts invocation propagates events into the transaction.

Reply:
ok